index.d.ts

```diff
@@ -473,6 +474,37 @@ declare namespace plugins {
     };
   }
 
+  /** @hidden */
+  interface Http2Client extends Http {
```

Comment:
Sounds good even though this means that other integrations (e.g. `hapi`, `express`, and others) will now inherit from `Http1...` instead of `Http..`.

In any case, I think we should do this in another PR.